src/blocks/posts/inspector.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+import {
+	PanelBody,
+	ToggleControl,
+	RangeControl,
+	QueryControls,
+	RadioControl,
+	BaseControl,
+	ButtonGroup,
+	Button,
+} from '@wordpress/components';
+
+const Inspector = props => {
+	const {
+		attributes,
+		activeStyle,
+		styleOptions,
+		onUpdateStyle,
+		setAttributes,
+		onUserModifiedColumn,
+		categoriesList,
+		postCount,
+		hasPosts,
+	} = props;
+
+	const {
+		order,
+		orderBy,
+		postFeedType,
+		postsToShow,
+		excerptLength,
+		displayPostDate,
+		displayPostContent,
+		columns,
+		listPosition,
+		imageSize,
+	} = attributes;
+
+	const isStackedStyle = ( 'stacked' === activeStyle.name );
+	const isHorizontalStyle = ( 'horizontal' === activeStyle.name );
+
+	const sizeOptions = [
+		{
+			value: 'w-1/7 h-1/7',
+			label: _x( 'Small', 'label for small size option' ),
+			shortName: _x( 'S', 'abbreviation for "Small" size' ),
+		},
+		{
+			value: 'w-1/7 sm:w-1/5 h-1/7 sm:h-1/5',
+			label: _x( 'Medium', 'label for medium size option' ),
+			shortName: _x( 'M', 'abbreviation for "Medium" size' ),
+		},
+		{
+			value: 'w-1/7 sm:w-1/3 h-1/7 sm:h-1/3',
+			label: _x( 'Large', 'label for large size option' ),
+			shortName: _x( 'L', 'abbreviation for "Large" size' ),
+		},
+		{
+			value: 'w-1/7 sm:w-1/3 md:w-1/2 h-1/7 sm:h-1/3 md:h-1/2',
+			label: _x( 'Extra Large', 'label for large size option' ),
+			shortName: _x( 'XL', 'abbreviation for "Large" size' ),
+		},
+	];
+
+	const settings = (
+		<PanelBody title={ __( 'Posts Settings' ) }>
+			<Fragment>
+				<ToggleControl
+					label={ __( 'Post Date' ) }
+					checked={ displayPostDate }
+					help={
+						displayPostDate ?
+							__( 'Showing the publish date.' ) :
+							__( 'Toggle to show the publish date.' )
+					}
+					onChange={ () => setAttributes( { displayPostDate: ! displayPostDate } ) }
+				/>
+				<ToggleControl
+					label={ __( 'Post Content' ) }
+					checked={ displayPostContent }
+					help={
+						displayPostContent ?
+							__( 'Showing the post content.' ) :
+							__( 'Toggle to show the post content.' )
+					}
+					onChange={ () => setAttributes( { displayPostContent: ! displayPostContent } ) }
+				/>
```

Comment:
We should use the post excerpt WordPress core already provides (and has associated hooks for other plugins/themes to use).